xarray/tests/test_units.py

```diff
@@ -2025,7 +2026,9 @@ def test_squeeze(self, dim, dtype):
         ),
         ids=repr,
     )
-    def test_computation(self, func, dtype):
+    def test_computation(self, func, dtype, compute_backend):
+        if compute_backend == "numbagg":
+            pytest.skip("numbagg functions are not supported by pint")
```

Comment:
I think we should move this exclusion from the test to the code itself. So if we're calculating on a pint array, we shouldn't use numbagg, and this test should still pass.

(@keewis can possibly provide some expertise on why the pint arrays don't work with some functions — lmk if it's an issue with numbagg, happy to work on that if so...)

Reply:
I could implement this in the quantiles method by checking if the input is a pint array or not using this:
https://github.com/pydata/xarray/blob/c9ba2be2690564594a89eb93fb5d5c4ae7a9253c/xarray/core/pycompat.py#L123

Or did you have something else in mind?

Reply:
Seems to work: https://github.com/pydata/xarray/pull/8684/commits/6130c7fa440dfc8af667118920a9ccdaaa17a0a6

Reply:
I'm a bit late, but this is a general issue with nested arrays: the individual layers need to be able to check (and maybe transform) the inputs, and `numbagg` should only have to deal with the lowest layer, like `numpy`, `cupy` or `sparse`.

The way this currently works is to teach `pint` about `numbagg` functions (through `__array_function__`, I think? `__array_namespace__` might support this, as well... how does `scipy` do this?), but I'm not sure where this should live (probably not in `pint`)